src/power.jl

```diff
@@ -0,0 +1,135 @@
+import Base: copy, get
+
+"""
+    IntervalMatrixPower{T}
+
+A wrapper for the matrix power that can be incremented.
+
+### Fields
+
+- `M`  -- the original matrix
+- `Mᵏ` -- the current matrix power, i.e., `M^k`
+- `k`  -- the current power index
+
+### Notes
+
+The wrapper should only be accessed using the interface functions.
+The internal representation (such as the fields) are subject to future changes.
+
+### Examples
+
+```jldoctest
+julia> A = IntervalMatrix([Interval(2.0) Interval(2.0, 3.0); Interval(0.0) Interval(-1.0, 1.0)])
+2×2 IntervalMatrix{Float64,Interval{Float64},Array{Interval{Float64},2}}:
+ [2, 2]   [2, 3]
+ [0, 0]  [-1, 1]
+
+julia> pow = IntervalMatrixPower(A);
+
+julia> increment!(pow)
+2×2 IntervalMatrix{Float64,Interval{Float64},Array{Interval{Float64},2}}:
+ [4, 4]  [2, 9]
+ [0, 0]  [0, 1]
+
+julia> increment(pow)
+2×2 IntervalMatrix{Float64,Interval{Float64},Array{Interval{Float64},2}}:
+ [8, 8]  [-1, 21]
+ [0, 0]   [-1, 1]
+
+julia> get(pow)
+2×2 IntervalMatrix{Float64,Interval{Float64},Array{Interval{Float64},2}}:
+ [4, 4]  [2, 9]
+ [0, 0]  [0, 1]
+
+```
+"""
+mutable struct IntervalMatrixPower{T}
+    M::IntervalMatrix{T}
+    Mᵏ::IntervalMatrix{T}
+    k::Int
+
+    function IntervalMatrixPower(M::IntervalMatrix{T}) where {T}
+        return new{T}(M, M, 1)
+    end
+
+    function IntervalMatrixPower(M::IntervalMatrix{T}, Mᵏ::IntervalMatrix{T},
+                                 k::Int) where {T}
```

Comment:
Add a check that `k >= 1`?

Reply:
:+1:
Although technically we could support `k = 0`, I will add `k >= 1`.